LSP is broken

Seems to have been caused by restructuring the overflow Db access. Will provide a detailed log about the error.

Error log:
```
15:55:14,688 ERROR CpgDocumentService Analysis error: 
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: The file is locked: nio:/codyze/graph-cache-overflow.bin [1.4.199/7]
        at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022) ~[?:?]
        at de.fraunhofer.aisec.crymlin.connectors.lsp.CpgDocumentService.analyze(CpgDocumentService.java:143) ~[codyze.jar:?]
        at de.fraunhofer.aisec.crymlin.connectors.lsp.CpgDocumentService.didOpen(CpgDocumentService.java:248) ~[codyze.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65) ~[org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113) [org.eclipse.lsp4j.jsonrpc-0.9.0.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalStateException: The file is locked: nio:/codyze/graph-cache-overflow.bin [1.4.199/7]
        at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at org.h2.mvstore.FileStore.open(FileStore.java:166) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at org.h2.mvstore.MVStore.<init>(MVStore.java:390) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at overflowdb.storage.OdbStorage.initializeMVStore(OdbStorage.java:142) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.storage.OdbStorage.getNodesMVMap(OdbStorage.java:131) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.storage.OdbStorage.allNodes(OdbStorage.java:122) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.initElementCollections(OdbGraph.java:112) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.<init>(OdbGraph.java:100) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.open(OdbGraph.java:81) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at de.fraunhofer.aisec.crymlin.connectors.db.OverflowDatabase.connect(OverflowDatabase.java:176) ~[codyze.jar:?]
        at de.fraunhofer.aisec.analysis.server.AnalysisServer.persistToODB(AnalysisServer.java:442) ~[codyze.jar:?]
        at de.fraunhofer.aisec.analysis.server.AnalysisServer.lambda$analyze$0(AnalysisServer.java:216) ~[codyze.jar:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705) ~[?:?]
        ... 1 more
Caused by: java.nio.channels.OverlappingFileLockException
        at sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229) ~[?:?]
        at sun.nio.ch.FileLockTable.add(FileLockTable.java:123) ~[?:?]
        at sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1154) ~[?:?]
        at org.h2.store.fs.FileNio.tryLock(FilePathNio.java:121) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at java.nio.channels.FileChannel.tryLock(FileChannel.java:1165) ~[?:?]
        at org.h2.mvstore.FileStore.open(FileStore.java:163) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at org.h2.mvstore.MVStore.<init>(MVStore.java:390) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343) ~[h2-mvstore-1.4.199.jar:1.4.199]
        at overflowdb.storage.OdbStorage.initializeMVStore(OdbStorage.java:142) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.storage.OdbStorage.getNodesMVMap(OdbStorage.java:131) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.storage.OdbStorage.allNodes(OdbStorage.java:122) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.initElementCollections(OdbGraph.java:112) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.<init>(OdbGraph.java:100) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at overflowdb.OdbGraph.open(OdbGraph.java:81) ~[overflowdb-tinkerpop3-0.128.jar:0.128]
        at de.fraunhofer.aisec.crymlin.connectors.db.OverflowDatabase.connect(OverflowDatabase.java:176) ~[codyze.jar:?]
        at de.fraunhofer.aisec.analysis.server.AnalysisServer.persistToODB(AnalysisServer.java:442) ~[codyze.jar:?]
        at de.fraunhofer.aisec.analysis.server.AnalysisServer.lambda$analyze$0(AnalysisServer.java:216) ~[codyze.jar:?]
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705) ~[?:?]
        ... 1 more
```

--- a/src/main/java/de/fraunhofer/aisec/analysis/server/AnalysisServer.java
+++ b/src/main/java/de/fraunhofer/aisec/analysis/server/AnalysisServer.java
@@ -439,7 +439,9 @@ public class AnalysisServer {
 		db.clearDatabase();
 
 		// connect
-		db.connect();
+		if (!db.isConnected()) {
+			db.connect();
+		}
 
 		// Persist the result
 		db.saveAll(result.getTranslationUnits());

